src/components/stake/StakeAmountSlider.vue

```diff
@@ -0,0 +1,577 @@
+        const alreadyStakedAmount = ref(props.stakedAmount);
+        const currentAmount = ref(alreadyStakedAmount.value);
+        const availableAmount = ref(activeAddressInfo.value?.balance);
+        const currentPercentage = computed({
+            get: () => (100 * currentAmount.value) / availableAmount.value!,
+            set: () => {
+                //
+            },
+        });
+        const alreadyStakedPercentage = ref(currentPercentage.value);
+        const alreadyStaked = ref(alreadyStakedAmount.value > 0);
+        const inputFocused = ref(false);
+        const currentFormattedAmount = computed(() =>
+            formatSpaceyNumber(currentAmount.value, NIM_MAGNITUDE));
+
+        const availableFormattedAmount = computed(() =>
+            formatSpaceyNumber(availableAmount.value!, NIM_MAGNITUDE));
+
+        const estimateTextWidth = (text: string, defaultSize: number, options = { ' ': 3 }) => {
+            let result = 0;
+            let special = 0;
+
+            Object.keys(options).forEach((key) => {
+                const found = text.split(key).length;
+                result += (options as {[key: string]: number})[key] * found;
+                special += found;
+            });
+            result += Math.max(2, (text.length - special)) * defaultSize;
+            return result;
+        };
+
+        const inputAmountWidth = computed(() => estimateTextWidth(currentFormattedAmount.value, 9) + 69);
+
+        const stakePercentage = computed(() => currentPercentage);
+        let containerBox:DOMRect;
+        let sliderBox:DOMRect;
+        let knobBox:DOMRect;
+        let amountBox:DOMRect;
+        let pivotPoint:Point = null;
+        let startSelection = 0;
+        let endSelection = 0;
+
+        const $container = ref<HTMLElement>(null);
+        const $knob = ref<HTMLElement>(null);
+        const $slide = ref<HTMLElement>(null);
+        const $backgroundOne = ref<HTMLElement>(null);
+        const $backgroundTwo = ref<HTMLElement>(null);
+        const $backgroundThree = ref<HTMLElement>(null);
+        const $backgroundFour = ref<HTMLElement>(null);
+        const $progressBar = ref<HTMLElement>(null);
+        const $percentText = ref<HTMLElement>(null);
+        const $stakedNIMText = ref<HTMLElement>(null);
+        const $stakedNIMAmount = ref<HTMLInputElement>(null);
+        const $dotIndicator = ref<HTMLElement>(null);
+
+        const atClick = (e: MouseEvent | TouchEvent) => {
+            e.preventDefault();
+
+            if (!active.value) {
+                active.value = true;
+                pivotPoint = extractInfo(e);
+                pivotPoint!.x -= $knob.value!.getBoundingClientRect().x;
+                window.addEventListener('mousemove', atMove);
+                window.addEventListener('touchmove', atMove);
+                window.addEventListener('mouseup', atEnd, { once: true });
+                window.addEventListener('touchend', atEnd, { once: true });
+            }
```

Comment:
Could avoid additional indentation via an early return:

```suggestion
            if (active.value) return;
            active.value = true;
            pivotPoint = extractInfo(e);
            pivotPoint!.x -= $knob.value!.getBoundingClientRect().x;
            window.addEventListener('mousemove', atMove);
            window.addEventListener('touchmove', atMove);
            window.addEventListener('mouseup', atEnd, { once: true });
            window.addEventListener('touchend', atEnd, { once: true });
```